### PR TITLE
DM-31213: Add github actions-based CI

### PR DESCRIPTION
**Pull request by spenczar**
Added Github actions to run lint, mypy, and pytest.

Set up integration tests which use real GCloud credentials/

Shortened some comments to under 80 characters to match LSST DM dev guide.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install
+        run: pip install -r <(curl https://raw.githubusercontent.com/lsst/linting/master/requirements.txt)
+
+      - name: Run linter
+        run: flake8 --ignore=E133,E226,E228 --max-line-length=110 --max-doc-length=79 --exclude bin,doc,**/*/__init.py,**/*/version.py,tests/.tests
```
Comment (jonathansick) [inline .github/workflows/test.yml]: Is this from the DM linting workflow template?

One thing that could be good long term is having a single source of truth for the flake8 configuration (and mypy too). You can put all this configuration into your `setup.cfg`, like: https://github.com/lsst/templates/blob/649807ecc7a7181f4bf6980946692b7ed9e09de1/project_templates/fastapi_safir_app/example/setup.cfg#L45-L58

That way this linting command will be consistent with `make lint`.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	pytest .
+
+.PHONY: integration-test
+integration-test:
+ifndef ALERTDB_TEST_GCP_PROJECT
+	$(error "ALERTDB_TEST_GCP_PROJECT environment variable must be set for integration tests")
+endif
+
+.PHONY: precommit
+precommit: lint type-check test
```
Comment (jonathansick) [inline Makefile]: I don't know how you feel about it, but we're also really into the [pre-commit](https://pre-commit.com) tool at SQuaRE. Here's are some pre-commit hooks we run by default: https://github.com/lsst/templates/blob/master/project_templates/fastapi_safir_app/example/.pre-commit-config.yaml

We don't have mypy in pre-commit (though you certainly could). We end up running those through separate tox enviroments.

**Review by jonathansick — APPROVED** [review]
Super. I just have a couple of comments that are mostly informational on approaches we've taken at SQuaRE, but are totally up to you.

File: .github/workflows/test.yml (hunk 1 above)
Comment (spenczar) [reply .github/workflows/test.yml]: Ah, yes, definitely better. Thanks!

File: Makefile (hunk 2 above)
Comment (spenczar) [reply Makefile]: Oh, nice. These are pretty slick! I've switched towards them, and dropped `make` entirely. Thanks!

**Pull request merged**